include/seastar/rpc/rpc_types.hh

```diff
@@ -73,6 +73,14 @@ struct client_info {
     typename std::add_const<T>::type& retrieve_auxiliary(const sstring& key) const {
         return const_cast<client_info*>(this)->retrieve_auxiliary<typename std::add_const<T>::type>(key);
     }
+    template <typename T>
+    std::optional<T> retrieve_auxiliary_opt(const sstring& key) const {
```

Comment:
Do we want to copy it? Existing functions return references. In C++ optional reference is a pointer, so my be this is what you want to return here.

Reply:
My initial implementation returned a pointer.
but I wanted to pass it as optional to `netw::msg_addr` constructor.
I can make it an optional in the caller too

Reply:
Yes, the interface should not depend on on use case.